language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -402,16 +433,19 @@ impl DependencyGraph {
                     entry.insert(ext_pkg);
                 }
 
-                // Seeing the same package in `extension` as in `self`: Not OK, even if their
-                // sources match, because supporting this case requires handling complicated edge
-                // cases (confirming that the sub-graph rooted at this package is also the same).
-                Entry::Occupied(entry) => {
-                    bail!(
-                        "Conflicting dependencies found:\n{0} = {1}\n{0} = {2}",
-                        ext_name,
-                        PackageWithResolverTOML(entry.get()),
-                        PackageWithResolverTOML(&ext_pkg),
-                    );
+                // Seeing the same package in `extension` is OK only if it has the same set of
+                // dependencies as the existing one.i
+                Entry::Occupied(_) => {
+                    let (self_deps, ext_deps) =
+                        pkg_deps_equal(ext_name, &self.package_graph, &ext_graph);
+                    if self_deps != ext_deps {
+                        bail!(
+                            "An already resolved package {} encountered during external resolution but with different dependencies:\n existing deps: {:?}\n external deps: {:?}",
```

Comment:
The error message should contain the following pieces of information:

- The list of mismatched dependencies on each side, rather than just the mismatched package being depended on. See `DependencyTOML` for a wrapper type that formats this kind of data (package name + dependency) in a human intelligible way.
- The external resolver being used.
- The package that's depending on these other packages.

Because otherwise there won't be sufficient context for someone to know where to look to fix the error.  Here's a sample error message to use as a template, that kind of matches what we had before, but with the additional information:

```
Conflicting dependencies found for 'ADep' during external resolution by 'resolver':
  Existing dependencies not found:
    ADepDep = { ... }

  New external dependencies:
    ADepDep = { ... }
```

And we should not display either of `Existing dependencies not found` or `New external dependencies` if the list is empty.